### PR TITLE
feat : cart and wishlist

### PR DESCRIPTION
**Pull request by ragini07**
1. Add to Cart 
2. Remove from Cart
3. Increment / decrement cart quantity
4. Add to Wishlist
5. Remove from Wishlist
6. Order summary based on cart products
( Login from test cred to check functionality)

### CODE REVIEW

File: src/Components/Cart/Cart.js [hunk 1]
```diff
@@ -1,9 +1,149 @@
...
+    
+    const [cartTotal , setCartTotal] = useState({
+        MRP : 0,
+        discount : 0,
+        TotalAmt : 0
+    })
+    const {userData , dispatchUserData } = useUser()
+    const {token} = useAuth()
+    const {cart , wishlist} = userData
+    console.log(cart)
```
Comment (rutvikpumak) [inline src/Components/Cart/Cart.js]: remove console

File: src/Components/Cart/Cart.js [hunk 2]
```diff
@@ -1,9 +1,149 @@
...
+             <>
+              <h2 className="center cart">My Cart</h2>
+             <div className='main-container'>
+           
+            <div className="cart-container">
+            <div className="left-container">
+                {/* <div className="address-card">
+                    <div className="detail">
+                        <h5>Deliver To :</h5><span>abc road,sakchi,JSR</span>
+                    </div>
```
Comment (rutvikpumak) [inline src/Components/Cart/Cart.js]: remove commented code

File: src/Components/Auth/Login.js [hunk 3]
```diff
@@ -52,9 +52,9 @@ function Login() {
             <button 
                 className="btn secondary sign-in-btn"
                 onClick={loginWithAdminHandler}>Login with Admin credential</button>
-            <span><a href="./forgotPassword.html">Forgot Password ?</a></span>
+            <span className='span-txt'><a href="./forgotPassword.html">Forgot Password ?</a></span>
```
Comment (anuj-kumary) [inline src/Components/Auth/Login.js]: Remove **href** from anchor tag

File: src/Components/Auth/auth.css [hunk 4]
```diff
@@ -35,7 +34,7 @@ h1{
     text-align: center;
    
 }
-span{
+.span-txt{
     line-height: 20px;
     font-size: 14px;
     margin:10px;
```
Comment (anuj-kumary) [inline src/Components/Auth/auth.css:40]: Try to be consistent about unit

File: src/Components/Cart/Cart.js (hunk 1 above)
Comment (anuj-kumary) [inline src/Components/Cart/Cart.js]: Remove **console.log**

File: src/Components/Cart/Cart.js [hunk 5]
```diff
@@ -1,9 +1,149 @@
...
+                    
+                    <button className="btn secondary"><a href="../AddressPage/address.html">Change</a></button>
+                </div> */}
```
Comment (anuj-kumary) [inline src/Components/Cart/Cart.js]: Remove unwanted comments

**Review by anuj-kumary — APPROVED** [review]

File: src/Context/auth-context.js [hunk 6]
```diff
@@ -12,6 +13,13 @@ export const AuthProvider = ({children}) => {
     const [token , setToken] = useState(authToken?.token)
     const [user , setUser] = useState(userData?.user)
 
+    console.log(token)
+    if(token){
+        console.log("in if to call user data from server")
```
Comment (rutvikpumak) [inline src/Context/auth-context.js]: remove consoles

File: src/Service/userAction.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    const {dispatchUserData} = useUser()
+ try {
+    const {data , status} = await axios.get('/api/user/cart',{
+        headers : {
+            authorization : token
+        }
+    })
+    if(status === 200) 
+         dispatchUserData({type : "SET_CART" , payload : data.cart})
+    console.log('get data from server', data)
```
Comment (rutvikpumak) [inline src/Service/userAction.js]: remove consoles

**Review by rutvikpumak — APPROVED** [review]
Remove all the consoles from your code. Use prettier to indent your code.
The rest of the things are well structured and clean code.

File: src/Components/Cart/Cart.js (hunk 5 above)
Comment (muskaanshah) [inline src/Components/Cart/Cart.js]: Remove commented code if not required

**Review by muskaanshah — APPROVED** [review]
Code looks good, functionality working as expected.
Just make the badge better on the icons. Good work 🌟

File: src/Components/Auth/Login.js (hunk 3 above)
Comment (ragini07) [reply src/Components/Auth/Login.js]: done

File: src/Components/Cart/Cart.js (hunk 5 above)
Comment (ragini07) [reply src/Components/Cart/Cart.js]: done

File: src/Components/Cart/Cart.js (hunk 2 above)
Comment (ragini07) [reply src/Components/Cart/Cart.js]: done

File: src/Service/userAction.js (hunk 7 above)
Comment (ragini07) [reply src/Service/userAction.js]: done

File: src/Components/Cart/Cart.js (hunk 5 above)
Comment (ragini07) [reply src/Components/Cart/Cart.js]: done

File: src/Components/Cart/Cart.js (hunk 1 above)
Comment (ragini07) [reply src/Components/Cart/Cart.js]: done

File: src/Components/Auth/auth.css (hunk 4 above)
Comment (ragini07) [reply src/Components/Auth/auth.css:40]: will try to be more consistent

File: src/Components/Cart/Cart.js (hunk 1 above)
Comment (ragini07) [reply src/Components/Cart/Cart.js]: done

**Pull request merged**